Sources/Mailgun/MailgunClient.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  MailgunClient.swift
+//  Mailgun
+//
+//  Created by Anthony Castelli on 11/14/16.
+//
+//
+
+import Console
+import Vapor
+import HTTP
+import Foundation
+import TurnstileCrypto
+import SMTP
+import Mail
+
+/**
+ Mailgun client
+ */
+public final class MailgunClient {
+    
+    static var defaultApiKey: String?
+    static var defaultDomain: String?
+    static var defaultClient: ClientProtocol.Type?
+    
+    var domain: String
+    var apiKey: String
+    var client: ClientProtocol
+    
+    public init(clientProtocol: ClientProtocol.Type, domain: String, apiKey: String) throws {
+        self.domain = domain
+        self.apiKey = apiKey
+        self.client = try clientProtocol.make(scheme: "https", host: "api.mailgun.net")
+    }
+    
+    public func send(_ emails: [MailgunEmail]) throws {
+        try emails.forEach { email in
+            let boundary = "vapor.mailgun.package.\(URandom().secureToken)"
+            let bytes = try self.createMultipartData(email.makeNode(), boundary: boundary)
+            let headers: [HeaderKey : String] = [
+                "Authorization": self.authorizationHeaderValue(self.apiKey),
+                "Content-Type": "multipart/form-data; boundary=\(boundary)"
+            ]
+            let response = try self.client.post(path: "/v3/\(self.domain)/messages", headers: headers, body: Body.data(bytes))
+            print(response)
+            switch response.status.statusCode {
+            case 200, 202: return
+            case 400: throw MailgunError.badRequest(try response.json?.extract())
+            case 401: throw MailgunError.unauthorized
+            case 500, 503: throw MailgunError.serverError
+            default: throw MailgunError.unexpectedServerResponse
+            }
+        }
+    }
+    
+    fileprivate func authorizationHeaderValue(_ apiKey: String) -> String {
+        let userPasswordString = "api:\(apiKey)"
+        let userPasswordData = userPasswordString.data(using: String.Encoding.utf8)
+        let base64EncodedCredential = userPasswordData!.base64EncodedString()
+        let authString = "Basic \(base64EncodedCredential)"
+        return authString
+    }
+    
+    fileprivate func createMultipartData(_ node: Node, boundary: String) -> Bytes {
+        var serialized = ""
+        
+        node.object?.forEach { key, value in
```

Comment:
In what situation could `node.object?` be `nil`? Would it make sense to guard against it and throw an error? The way it's currently implemented, if the node provided isn't an object, then the function will still return bytes and the application won't know anything went wrong until Mailgun itself returns an error.